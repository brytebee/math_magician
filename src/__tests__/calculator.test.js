import ReactDOM from 'react-dom';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Calculator from '../components/calculator';

afterEach(cleanup);

describe('Snapshot of Calculator component', () => {
  it('Screen displays the Calculator component', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Calculator />, div);
  });

  it('Take a snapshot of the Calculator component', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
