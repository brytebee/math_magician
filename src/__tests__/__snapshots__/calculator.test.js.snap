// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Calculator component Take a snapshot of the Calculator component 1`] = `
<div
  className="d-flex"
>
  <div
    className="container"
  >
    <div
      className="button result"
    >
      <p
        className="result-text"
      >
        0
      </p>
    </div>
    <div
      className="d-flex"
    >
      <button
        className="button flex-child"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="button flex-child"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="button flex-child"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="button flex-child operator"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="d-flex"
    >
      <button
        className="button flex-child"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="button flex-child"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="button flex-child"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="button flex-child operator"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="d-flex"
    >
      <button
        className="button flex-child"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="button flex-child"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="button flex-child"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="button flex-child operator"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="d-flex"
    >
      <button
        className="button flex-child"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="button flex-child"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="button flex-child"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="button flex-child operator"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="d-flex"
    >
      <button
        className="button big-child"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="button flex-child"
        type="button"
      >
         
        .
         
      </button>
      <button
        className="button flex-child operator"
        onClick={[Function]}
        type="button"
      >
         
        =
         
      </button>
    </div>
  </div>
</div>
`;
